mangaki/mangaki/wrappers/anilist.py

```diff
@@ -414,14 +502,177 @@ def get_user_list(self,
+        titles.update({entry.english_title: ('english', 'main')})
+        titles.update({entry.japanese_title: ('japanese', 'official')})
+
+        anime_type = entry.media_type if entry.work_type == AniListWorks.animes else ''
+        manga_type = entry.media_type if entry.work_type == AniListWorks.mangas else ''
+
+        studio = None
+
+        category = category_map.get(entry.work_type, None)
+
+        work_present_in_db = Work.objects.filter(
+            category__slug=entry.work_type.value,
+            title__in=titles
+        )
+
+        # If one of this entry's title match in the database, skip it
+        if work_present_in_db:
+            continue
+
+        # Link Studio and Work [AniListRichEntry only]
+        if isinstance(entry, AniListRichEntry) and entry.studio:
+            studio, created = Studio.objects.get_or_create(title=entry.studio)
+
+        # Create the Work entry in the database
+        work = Work.objects.create(
+            category=category,
+            title=entry.title,
+            ext_poster=entry.poster_url,
+            nsfw=entry.is_nsfw,
+            date=entry.start_date,
+            end_date=entry.end_date,
+            ext_synopsis=(entry.description if entry.description else ''),
+            nb_episodes=(entry.nb_episodes or entry.nb_chapters),
+            anime_type=anime_type,
+            manga_type=manga_type,
+            studio=studio
+        )
+
+        # Build genres for this Work
+        for genre_title in entry.genres:
+            genre, created = Genre.objects.get_or_create(title=genre_title)
+            work.genre.add(genre)
+
+        # Create WorkTitle entries in the database for this Work
+        current_work_titles = [
+            WorkTitle(
+                work=work,
+                title=title,
+                ext_language=language_map[language],
+                language=language_map[language].lang,
+                type=title_type
+            ) for title, (language, title_type) in titles.items()
+        ]
+        WorkTitle.objects.bulk_create(current_work_titles)
+
+        # Build RelatedWorks (and add those Works too) [AniListRichEntry only]
+        if isinstance(entry, AniListRichEntry) and entry.relations:
+            new_relations = []
+
+            for work_related, relation_type in entry.relations:
+                added_work = insert_works_into_database_from_anilist([work_related])
+                if not added_work:
+                    continue
+
+                new_relations.append(
+                    RelatedWork(
+                        parent_work=work,
+                        child_work=added_work[0],
+                        type=relation_type.name
+                    )
+                )
+
+            RelatedWork.objects.bulk_create(new_relations)
+
+        # Build Artist and Staff [AniListRichEntry only]
+        if isinstance(entry, AniListRichEntry) and entry.staff:
```

Comment:
Relying on instance types is more or less okay, but when you can, it's preferable to use [duck typing](https://fr.wikipedia.org/wiki/Duck_typing) [1], just add an `is_rich` property on both types and test for it, that'll make it easier to extend afterwards.

[1]: Okay it's not exactly Duck Typing, because we are cheating using `is_rich`, but it's close.